controllers/llamastackdistribution_controller.go

```diff
@@ -1190,34 +1227,315 @@ func (r *LlamaStackDistributionReconciler) getConfigMapHash(ctx context.Context,
+	}
+
+	concatenated := builder.String()
+	c.logger.Info("Successfully gathered CA bundle data",
+		"totalCertificates", c.certificateCount,
+		"totalSize", len(concatenated))
+
+	return concatenated, nil
+}
+
+func (r *LlamaStackDistributionReconciler) gatherExplicitCABundle(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution, collector *certificateCollector) error {
+	if !r.hasCABundleConfigMap(instance) {
+		return nil
+	}
+
+	configMapNamespace := r.getCABundleConfigMapNamespace(instance)
 	configMap := &corev1.ConfigMap{}
 	err := r.Get(ctx, types.NamespacedName{
 		Name:      instance.Spec.Server.TLSConfig.CABundle.ConfigMapName,
 		Namespace: configMapNamespace,
 	}, configMap)
 	if err != nil {
-		return "", err
+		return fmt.Errorf("failed to get CA bundle ConfigMap %s/%s: %w",
+			configMapNamespace, instance.Spec.Server.TLSConfig.CABundle.ConfigMapName, err)
 	}
 
-	// Create a content-based hash that will change when the ConfigMap data changes
-	// Include information about which keys are being used
-	var keyInfo string
-	if len(instance.Spec.Server.TLSConfig.CABundle.ConfigMapKeys) > 0 {
-		keyInfo = fmt.Sprintf("-%s", strings.Join(instance.Spec.Server.TLSConfig.CABundle.ConfigMapKeys, ","))
+	keysToProcess := instance.Spec.Server.TLSConfig.CABundle.ConfigMapKeys
+	if len(keysToProcess) == 0 {
+		keysToProcess = []string{DefaultCABundleKey}
+	}
+
+	return r.processConfigMapKeys(configMap, keysToProcess, configMapNamespace, instance.Spec.Server.TLSConfig.CABundle.ConfigMapName, collector)
+}
+
+func (r *LlamaStackDistributionReconciler) gatherODHCABundle(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution, collector *certificateCollector) error {
+	configMap, keys, err := r.detectODHTrustedCABundle(ctx, instance)
+	if err != nil {
+		// Log but don't fail - ODH bundle is optional
+		collector.logger.V(1).Info("Could not detect ODH trusted CA bundle", "error", err)
+		return nil
+	}
+	if configMap == nil || len(keys) == 0 {
+		return nil
+	}
+
+	return r.processODHConfigMapKeys(configMap, keys, collector)
+}
+
+func (r *LlamaStackDistributionReconciler) processConfigMapKeys(configMap *corev1.ConfigMap, keys []string, namespace, name string, collector *certificateCollector) error {
+	for _, key := range keys {
+		data, exists := configMap.Data[key]
+		if !exists {
+			return fmt.Errorf("failed to find CA bundle key '%s' in ConfigMap %s/%s", key, namespace, name)
+		}
+
+		certs, size, count, err := extractValidCertificates([]byte(data), key)
+		if err != nil {
+			return fmt.Errorf("failed to process CA bundle key '%s' from ConfigMap %s/%s: %w", key, namespace, name, err)
+		}
+
+		if err := collector.add(certs, size, count, configMap.Name, key); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *LlamaStackDistributionReconciler) processODHConfigMapKeys(configMap *corev1.ConfigMap, keys []string, collector *certificateCollector) error {
+	for _, key := range keys {
```

Comment:
Is there a reason to explicitly have odh specific function here instead of just using the above function?

Reply:
This function was intended to be different from `processConfigMapKeys` in a couple ways.

For `processConfigMapKeys`, if the user specifies a ConfigMap that doesn't exist or a Key that has formatting issues, the deployment should fail until those issues are resolved by the user.

If the ODH bundle has some kind of issue, it may not be something that can be resolved by the user if they don't also manage the ODH instance. So the `processODHConfigMapKeys` function only warns them about it and then continues.